tools/diagnostics/nscapture

```diff
@@ -0,0 +1,782 @@
+    time.sleep(1)
+
+    logger.setLevel(args.log_level)
+    logger.debug("args: %s", args)
+    find_node_name(args)
+
+    atexit.register(clean_up)
+
+
+def build_archive_cmd(output_file, files):
+    cmd = ['tar']
+    flags = 'cfz'
+    if logger.level == logging.DEBUG:
+        flags += 'v'
+    cmd.append(flags)
+    cmd.append(output_file)
+    cmd.append('--show-transformed-names')
+    cmd.extend(["-C", "/"])
+
+    for entry in files:
+        fs_path_pattern = entry.abs_fs_path.strip('/')
+        archive_path_pattern = entry.archive_path.lstrip('/')
+        if archive_path_pattern:
+            archive_path_pattern += '/'  # Add slash for directories otherwise leave blank
+
+        if entry.include:
+            cmd.append('--xform')
+            cmd.append("s|%s/|%s|1" % (fs_path_pattern, archive_path_pattern))
+            for file in entry.include:
+                f_path = os.path.join(entry.abs_fs_path, file)
+                f_path = os.path.abspath(f_path)
+                f_path = f_path.lstrip('/')
+                cmd.append(f_path)
+        else:
+            cmd.append('--xform')
+            cmd.append("s|%s|%s|1" % (fs_path_pattern, archive_path_pattern))
+            f_path = entry.abs_fs_path.strip('/')
+            cmd.append(f_path + '/')  # Add slash to end for sed expression matching
+
+    return cmd
+
+
+# ***************
+# *  Running the archive command
+# ***************
+def run_cmd(cmd):
+    logger.info("Running archive command")
+    try:
+        subprocess.run(cmd, check=True)
+        # os.system(' '.join(cmd))
+    except Exception as e:
+        logger.error("Unable to complete archive command: %s", str(e))
+    else:
+        logger.info("Archive command complete")
+
+
+def do_capture(args, files):
+    output_file = os.path.join(args.output_dir, capture_file_name(args) + '.tar.gz')
+    logger.info("Writing archive to '%s'" % output_file)
+
+    logger.info("Building archive command")
+    cmd = build_archive_cmd(output_file, files)
+    logger.debug("Archive command to be run:")
+    logger.debug(' '.join(cmd))
+
+    if args.dry_run:
+        logger.info("Dry-run")
+        logger.info("Collected following files:")
+        _log_capture(files)
+        return
+    else:
+        # if os.getuid() is not 0:
+        # TODO Need to be more clever, only need to be super user when on a bare metal box
+        # logger.error("Must be run as super user (sudo) to capture all files.")
+        # raise Exception("Not Super User")
+        run_cmd(cmd)
+
+
+def main(args):
+    try:
+        init(args)
+    except:
+        logger.error('Unable to initialize script')
+        raise
+
+    try:
+        files = collect_capture_files(args)
+    except:
+        logger.error('Unable to collect the files for this capture')
+        raise
+
+    try:
+        do_capture(args, files)
+    except:
+        logger.error('Unable to build capture archive')
+        raise
+
+
+# ***************
```

Comment:
From Alex:
We use pytest for testing. Can we switch to pytest to execute the test during CI?

Reply:
And please move tests to a separate folder (like in other places in code)

Reply:
nscapture was written as a stand-alone script. It is not a python module. It is outside the view of pytest. I choose unittest so there was no dependency to run the test. Also, unittest was easier to run via a command line option (-t). I'm welcome to discuss how to do this better but simply putting them in a directory and using pytest would not accomplish my portable stand-alone CLI script requirement requirements.

The tests are exercised in the CI system in indy_node/test/tools/test_nsrepay.py

Reply:
OK. We can create a ticket for more enhancements.